logstash_async/transport.py

```diff
@@ -176,3 +316,173 @@ def send(self, events, use_logging=False):
+        :rtype: None or str
+        """
+        return self.__username
+
+    @username.setter
+    def username(self, name: Union[None, str]) -> None:
+        self.__username = name
+
+    @property
+    def password(self) -> Union[None, str]:
+        """password
+        :param word: The password of the user
+        :type word: str
+        :return: The current password or None
+        :rtype: None or str
+        """
+        return self.__password
+
+    @password.setter
+    def password(self, word: Union[None, str]) -> None:
+        self.__password = word
+
+    @property
+    def url(self) -> str:
+        """The URL of the logstash pipeline based on the hostname, the port and
+        the TLS usage.
+
+        :return: The URL of the logstash pipeline
+        :rtype: str
+        """
+        protocol = "http"
+        if self.ssl_enable:
+            protocol = "https"
+        return f"{protocol}://{self.host}:{self.port}"
+
+    def __auth(self):
+        """The authentication method for the logstash pipeline. If the username
+        or the password is not set correctly it will return None.
+
+        :return: A HTTP basic auth object or None
+        :rtype: HTTPBasicAuth
+        """
+        if self.username is None or self.password is None:
+            return None
+        return HTTPBasicAuth(self.username, self.password)
+
+    @property
+    def headers(self) -> dict:
+        """The headers of the HTTP request
+
+        :return: A dictionary with HTTP header fields
+        :rtype: dict
+        """
+        request_headers = {
+            "Content-Type": "text/plain",
+        }
+        if self.codec is None:
+            return request_headers
+        if self.codec.lower == "json":
+            request_headers["Content-Type"] = "application/json"
+        return request_headers
+
+    def encode(self, data: bytes) -> str:
+        """Encodes the data based on the chosen codec
+
+        :param data: The input data
+        :type data: bytes
+        :return: The encoded data
+        :rtype: str
+        """
+        if self.codec is None or data is None:
+            return data
+        string = data.decode("utf-8")
+        if self.codec.lower() == "json":
+            return json.loads(string)
+        return string
+
+    def close(self) -> None:
+        """The HTTP connection does not need to be closed because it's
+        stateless.
+        """
+        if self.__session is not None:
+            self.__session.close()
+
+    def __backoff(self, attempt, cap: int = 3000, base: int = 10):
+        return random.randrange(0, min(cap, base * 2 ** attempt))
+
+    def send(self, events: list, **kwargs: dict) -> None:
+        """Send events to the logstash pipeline
+
+        :param events: A list of events
+        :type events: list
+        :param use_logging: Not used!
+        :type use_logging: bool
+        """
+        self.__session = requests.Session()
+        for event in events:
```

Comment:
Does the HTTP input support receiving multiple events at once?
This would speed up the transmission a lot. Currently the code would perform one HTTP request per event.

Reply:
You are right, multiline events are possible. I will add this and add some chunk mechanism based on the post_max_size.